src/common/streak.ts

```diff
@@ -0,0 +1,75 @@
+import { DayOfWeek, DEFAULT_WEEK_START } from './date';
+import {
+  Day,
+  FREEZE_DAYS_IN_A_WEEK,
+  STREAK_RECOVERY_MAX_GAP_DAYS,
+} from './users';
+
+interface DaysProps {
+  day: number;
+  firstDayOfWeek: Day;
+  lastDayOfWeek: Day;
+  lastView: Date;
+}
+
+const getNextDay = (day: number) => (day === 6 ? 0 : day + 1);
+const getPreviousDay = (day: number) => (day === 0 ? 6 : day - 1);
+
+export const getAllowedDays = ({
+  day: dayToday,
+  firstDayOfWeek,
+  lastDayOfWeek,
+  lastView,
+}: DaysProps) => {
+  const lastViewDay = lastView.getDay();
+  const dayTwo = getNextDay(firstDayOfWeek);
```

Comment:
What does dayTwo mean, this is the first unclear thing to me?

Reply:
Day two of the weekday, similar to the other comment, to make it work across different start of the week. Rather than having the same amount of conditions be copy-pasted to support different start of the week.

Reply:
So based on the weekend, we would have here either Tuesday for a week that starts on Monday, or Monday for a week that starts on Saturday right?

I think it's clearer if we call this `secondDayOfWeek` or `weekDayTwo`

Reply:
You mean starts on Sunday? Yes.

Sure, we can always have better variable names.

Reply:
> You mean starts on Sunday? Yes.


yup my mistake there, just want to make sure the naming explain itself